distributed/shuffle/_scheduler_extension.py

```diff
@@ -63,6 +62,13 @@ def __init__(self, scheduler: Scheduler):
     def shuffle_ids(self) -> set[ShuffleId]:
         return set(self.states)
 
+    async def barrier(self, id: ShuffleId, run_id: int) -> None:
+        shuffle = self.states[id]
```

Comment:
We _could_ check the run_id here to avoid the broadcasting in the first place, but if it is stale, workers will refuse `shuffle_inputs_done`.